tests/api/v1/samples/upsertBulk.js

```diff
@@ -211,6 +211,111 @@ describe('api: POST ' + path, () => {
     });
   });
 });
+
+describe('api: POST ' + path + ' upsert bulk when sample ' +
+  'already exists', () => {
+  let token;
+  let aspect1Id = null;
+  before((done) => {
+    tu.createToken()
+    .then((returnedToken) => {
+      token = returnedToken;
+      done();
+    })
+    .catch((err) => done(err));
+  });
+
+  before((done) => {
+    Aspect.create({
+      isPublished: true,
+      name: `${tu.namePrefix}Aspect1`,
+      timeout: '30s',
+      valueType: 'NUMERIC',
+      criticalRange: [0, 1],
+    })
+    .then((aspect1) => {
+      aspect1Id = aspect1.id;
+      Aspect.create({
+        isPublished: true,
+        name: `${tu.namePrefix}Aspect2`,
+        timeout: '10m',
+        valueType: 'BOOLEAN',
+        okRange: [10, 100],
+      });
+    })
+    .then(() => Subject.create({
+      isPublished: true,
+      name: `${tu.namePrefix}Subject`,
+    }))
+    .then((subject) => Sample.create({
+      subjectId: subject.id,
+      aspectId: aspect1Id
+    }))
+    .then(() => done())
+    .catch((err) => done(err));
+  });
+
+  after(u.forceDelete);
+  after(tu.forceDeleteUser);
+
+  it('check that duplication of sample is not happening', (done) => {
+    api.post(path)
+    .set('Authorization', token)
+    .send([{
+      name: `${tu.namePrefix}Subject|${tu.namePrefix}Aspect1`,
+      value: '6',
+    }])
+    .then(() => {
+      api.get('/v1/samples?name=' +
+        `${tu.namePrefix}Subject|${tu.namePrefix}Aspect1`)
+      .expect((res) => {
+        setTimeout(() => {
```

Comment:
I don't understand the "setTimeout" use here... you can use chai-as-promised "should.eventually..."